R/h_bracket.R

```diff
@@ -0,0 +1,16 @@
+
+#' h_bracket
+#'
+#' Generic function that creates a character string oriented by curly brackets
+#'
+#' @param x An Object
+#' @export
+h_bracket <- function(x) {
+    if (any(nchar(x) >= 1, length(x) > 1)) {
+        paste0(
+            "{\n",
+            paste0(x, collapse = "\n"),
+            "\n}\n"
+        )
+    }
```

Comment:
@G-Kazantzidis , Is this correct ? At the moment if  the `if()` condition is not met the function returns literally nothing ?

Reply:
This is correct. The scenario that this is False is when we have a the character vector like `""`. In that case, the section is empty to there is no need to create it ( eg `generated quantities { }`.

Reply:
However if the user provides a vector of two empty strings ( eg `stanmodule(model = c("","") )` ), it will return `"{\n \n}\n"`.

Reply:
ah ok,  can we make this explicit though i.e.  adding a return outside of the `if()` condition that explicitly returns nothing.    `return(x)`  or  `return("")`  or  `return(invisible()`  at least this way it is clear it is a conscious decision. Also perhaps a unit test to test that this assumption is working as intended.

Reply:
The difference is that if we `return(x)` or `return("")`, an empty line is going to be added to the stan code. Whereas like this, no line is going to be added if a section slot is empty. I understand it is not a major difference.